---
output: html_document
---

```{r setup, include=FALSE}
# general packages
library(tidyverse)
library(conflicted)
library(magrittr)

# plotting packages
library(viridis)
library(wesanderson)

# resolve package conflicts
filter <- dplyr::filter

knitr::opts_chunk$set(echo = TRUE)
```

## Figure 1: Affect of Microbiome and Inflammation (IL10KO) on Methylation

Code for Figure 1 for the paper The Microbiome Modulates DNA Methylation in Mouse Colonic Mucosa

### Wrangle Data

These tables were created by a former graduate student in the lab, Lauren Cole, who performed t-tests for experimental condition vs. experimental condition in Excel and annotated the result tables with CpG location information.

-

Read in the data

```{r}
# get file names
lauren_files <- list.files(path = '../../../data/lauren_volc_tbls', 
                           pattern = '*.csv', 
                           full.names = T)

# read files in recursively
tibble(filename = lauren_files) %>%
# read in files recursively, adding on filename
  mutate(file_contents = map(filename, ~ read_csv(.))) %>%
# expand files into skinny table
  unnest() %>%
# select only needed columns (many extra annotation columns in these tables)
  select(filename, ID, Chromosome, Mm9_SmaI_annotation_Position, 
         Ave_Y, O.Y, AvgWT, KO.WT, AvgGF, SPF.GF, AvgN, AOM.N, ttest2, pttest2, 
         inCpG, cgi.shore) -> lauren_data
```

Wrangle data

```{r}
lauren_data %>%
# separate the filename into component folders and parts with the ultimate goal
# of using the filename to create a sample ID
  separate(filename, into = c('folder1', 'folder2', 'folder3', 'folder4', 
                              'folder5', 'filename'), 
           sep = '/') %>%
# drop the folder names because I don't need them
  select(-(folder1:folder5)) %>% 
# separate the file names into their main bodies and file extensions
  separate(filename, into = c('file_prefix', 'file_extension'), sep = -4) %>% 
# drop the file extensions because I don't need them; I know they're all CSVs
  select(-file_extension) %>%
# select everything from the filename up until the first _ and use that as the
# sample ID
  mutate(sample_id = str_extract(file_prefix, "[^_]+")) %>%
# replace NAs, which are only in the methylation difference columns; this way I
# can add them together to make a unified methylation difference column rather 
# than using a complicate if or case_when statement
  replace(is.na(.), 0) %>%
# add the methylation difference columns together to make a unified methylation
# difference column
  mutate(meth_diff = O.Y + KO.WT + SPF.GF + AOM.N,
# add a categorical column annotating the direction of methylation change for
# easier counting later
         baseline_meth = Ave_Y + AvgWT + AvgGF + AvgN,
         baseline_anno = factor(case_when(baseline_meth <= 20 ~ 'Me < 20',
                                   baseline_meth > 20 & 
                                     baseline_meth < 80 ~ '20 < Me < 80',
                                   baseline_meth >= 80 ~ 'Me > 80'),
                                levels = c('Me < 20', '20 < Me < 80', 
                                           'Me > 80')),
         location_anno = case_when(inCpG == 'Y' ~ 'CGI',
                                   cgi.shore == 1 ~ 'CGI shore',
                                   TRUE ~ 'other'),
         meth_change_0 = factor(ifelse(meth_diff < 0, 'hypo', 'hyper'), 
                              levels = c('hypo', 'hyper')),
         meth_change_5 = factor(case_when(meth_diff <= -5 ~ 'hypo', 
                                          meth_diff >= 5 ~ 'hyper',
                                          TRUE ~ 'no_change'), 
                              levels = c('hypo', 'hyper', 'no_change')),
# add a categorical column annotating whether the pvalue is significant or not
# for easier counting later
         significance = ifelse(ttest2 < 0.05, 'sig', 'notsig'),
# add a categorical column annotating color for easy coloring on volcano plots
         volc_color = ifelse(abs(meth_diff) >= 5 & ttest2 < 0.05, 
                             'sig', 'notsig')) %>%
# keep only the columns I need and rename some for better intuition/easier use 
# down the line
  select(sample_id, sma_id = ID, 
         chrom = Chromosome, pos = Mm9_SmaI_annotation_Position, 
         meth_diff, pvalue = ttest2, 
         logpvalue = pttest2, meth_change_0, 
         meth_change_5, significance, 
         volc_color, baseline_meth, 
         baseline_anno, location_anno) -> lauren_data_wrangled
```

### Volcano Plots

Volcano plots, stacked bar plots of significant CpG locations, stacked bar plot of whether sites were were significant for the microbiome, IL10KO, or both

#### Figure A: **SPF - GF** Volcano plot

Methylation differences between specific pathogen free mice and germ free mice. Contribution from having a microbiome.

```{r, fig.width = 8, fig.height = 8}
### volcano plot annotation
lauren_data_wrangled %>% 
  filter(sample_id == 'spf-gf') %>%
  add_count(name = 'sample_count') %>%
  group_by(meth_change_5, volc_color, sample_count) %>% 
  count() %>% 
  ungroup() %>%
  filter(volc_color != 'notsig') %>%
  mutate(meth_diff = c(-30, 30), 
         logpvalue = 9, 
         label = paste0(n, 
                        ', ', 
                        round(((n / sample_count) * 100), 1), 
                        '%')) -> spfgf_labels

### plot
lauren_data_wrangled %>%
  filter(sample_id == 'spf-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue)) +
  geom_point(aes(color = volc_color), alpha = 0.5) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_text(data = spfgf_labels, aes(label = label), size = 10) +
  geom_vline(xintercept = c(-5, 5), color = 'gray40', linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), color = 'gray40', linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (SPF - GF)',
       y = '-Log10 PValue',
       title = 'A') +
  coord_cartesian(xlim = c(-50, 50),
                  ylim = c(0, 10)) +
  theme_classic(base_size = 20) +
  theme(legend.position = 'none')

### save
ggsave('spf-gf_volcano.png', width = 8, height = 8)
ggsave('spf-gf_volcano.pdf', width = 8, height = 8)
ggsave('spf-gf_volcano.tiff', width = 8, height = 8)
```

#### Figure B: **GFIL10KO - GF*** Volcano plot 

Methylation differences between germ free mice with IL10KO and germ free mice. Contribution from the IL10KO.

```{r, fig.width = 8, fig.height = 8}
### volcano plot annotation
lauren_data_wrangled %>% 
  filter(sample_id == 'gfko-gf') %>%
  add_count(name = 'sample_count') %>%
  group_by(meth_change_5, volc_color, sample_count) %>% 
  count() %>% 
  ungroup() %>%
  filter(volc_color != 'notsig') %>%
  mutate(meth_diff = c(-30, 30), 
         logpvalue = 9, 
         label = paste0(n, 
                        ', ', 
                        round(((n / sample_count) * 100), 1), 
                        '%')) -> gfkogf_labels

### plot
lauren_data_wrangled %>%
  filter(sample_id == 'gfko-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue)) +
  geom_point(aes(color = volc_color), alpha = 0.5) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_text(data = gfkogf_labels, aes(label = label), size = 10) +
  geom_vline(xintercept = c(-5, 5), color = 'gray40', linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), color = 'gray40', linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (GFko - GF)',
       y = '-Log10 PValue',
       title = 'B') +
  coord_cartesian(xlim = c(-50, 50),
                  ylim = c(0, 10)) +
  theme_classic(base_size = 20) +
  theme(legend.position = 'none')

### save
ggsave('gfko-gf_volcano.png', width = 8, height = 8)
ggsave('gfko-gf_volcano.pdf', width = 8, height = 8)
ggsave('gfko-gf_volcano.tiff', width = 8, height = 8)
```

#### Figure C: **SPFKO - GF** Volcano Plot

Methylation differences between specific pathogen free mice with IL10KO and germ free mice. Synergy between the two conditions

```{r, fig.width = 8, fig.height = 8}
### volcano plot annotation
lauren_data_wrangled %>% 
  filter(sample_id == 'spfko-gf') %>%
  add_count(name = 'sample_count') %>%
  group_by(meth_change_5, volc_color, sample_count) %>% 
  count() %>% 
  ungroup() %>%
  filter(volc_color != 'notsig') %>%
  mutate(meth_diff = c(-30, 30), 
         logpvalue = 9, 
         label = paste0(n, 
                        ', ', 
                        round(((n / sample_count) * 100), 1), 
                        '%')) -> spfkogf_labels

### plot
lauren_data_wrangled %>%
  filter(sample_id == 'spfko-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue)) +
  geom_point(aes(color = volc_color), alpha = 0.5) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_text(data = spfkogf_labels, aes(label = label), size = 10) +
  geom_vline(xintercept = c(-5, 5), color = 'gray40', linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), color = 'gray40', linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (SPFko - GF)',
       y = '-Log10 PValue',
       title = 'C') +
  coord_cartesian(xlim = c(-50, 50),
                  ylim = c(0, 10)) +
  theme_classic(base_size = 20) +
  theme(legend.position = 'none')

### save
ggsave('spfko-gf_volcano.png', width = 8, height = 8)
ggsave('spfko-gf_volcano.pdf', width = 8, height = 8)
ggsave('spfko-gf_volcano.tiff', width = 8, height = 8)
```

### Stacked CpG Location Bar Plots

Count number of sites detected in each sample.

```{r}
lauren_data_wrangled %>%
  group_by(sample_id) %>%
  count() %>%
  ungroup() %>%
  rename(num_sites_detected = n) %>%
  spread(sample_id, num_sites_detected) -> num_sites_detected
```

#### Figure D: **SPF - GF** Stacked Bar Plots

Location of significant CpG sites stratified by baseline methylation from the volcano plot between specific pathogen free mice and germ free mice. 

```{r, fig.width = 8, fig.height = 8}
### filter / wrangle
lauren_data_wrangled %>%
# filter for sites in the given condition that significantly changed
  filter(sample_id == 'spf-gf',
         significance == 'sig',
         meth_change_5 != 'no_change') %>%
# calculate what proportion of the total sites the significant sites are,
# grouped by their location in DNA
  group_by(significance, meth_change_5, location_anno, baseline_anno) %>%
  count() %>%
  ungroup() %>%
  mutate(proportion = (n / num_sites_detected$`spf-gf`)) %>%

### plot
ggplot(aes(x = meth_change_5, y = proportion, fill = location_anno)) +
  geom_bar(stat = 'identity') +
  scale_fill_viridis_d(direction = -1, option = 'cividis') +
  facet_wrap(~ baseline_anno) +
  labs(x = 'Direction of Methylation Change',
       y = 'Proportion of Total Detectable Sites',
       fill = '',
       title = 'D') +
  coord_cartesian(ylim = c(0, 0.06)) +
  theme_classic(base_size = 20) 

### save
ggsave('spf-gf_stacked_bar.png', width = 8, height = 8)
ggsave('spf-gf_stacked_bar.pdf', width = 8, height = 8)
ggsave('spf-gf_stacked_bar.tiff', width = 8, height = 8)
```

#### Figure E: **GFKO - GF** Stacked Bar Plots

Location of significant CpG sites stratified by baseline methylation from the volcano plot between germ free mice with IL10KO and germ free mice. 

```{r, fig.width = 8, fig.height = 8}
### filter / wrangle
lauren_data_wrangled %>%
# filter for sites in the given condition that significantly changed
  filter(sample_id == 'gfko-gf',
         significance == 'sig',
         meth_change_5 != 'no_change') %>%
# calculate what proportion of the total sites the significant sites are,
# grouped by their location in DN
  group_by(significance, meth_change_5, location_anno, baseline_anno) %>%
  count() %>%
  ungroup() %>%
  mutate(proportion = (n / num_sites_detected$`gfko-gf`)) %>%

### plot
ggplot(aes(x = meth_change_5, y = proportion, fill = location_anno)) +
  geom_bar(stat = 'identity') +
  scale_fill_viridis_d(direction = -1, option = 'cividis') +
  facet_wrap(~ baseline_anno) +
  labs(x = 'Direction of Methylation Change',
       y = 'Proportion of Total Detectable Sites',
       fill = '',
       title = 'E') +
  coord_cartesian(ylim = c(0, 0.06)) +
  theme_classic(base_size = 20)

### save
ggsave('gfko-gf_stacked_bar.png', width = 8, height = 8)
ggsave('gfko-gf_stacked_bar.pdf', width = 8, height = 8)
ggsave('gfko-gf_stacked_bar.tiff', width = 8, height = 8)
```

### Figure F: **SPFKO - GF** Stacked Bar Plot

Location of significant CpG sites stratified by baseline methylation from the volcano plot between specific pathogen free mice with IL10KO and germ free mice. 

```{r, fig.width = 8, fig.height = 8}
### filter / wrangle
lauren_data_wrangled %>%
# filter for sites in the given condition that significantly changed
  filter(sample_id == 'spfko-gf',
         significance == 'sig',
         meth_change_5 != 'no_change') %>%
# calculate what proportion of the total sites the significant sites are,
# grouped by their location in DN
  group_by(significance, meth_change_5, location_anno, baseline_anno) %>%
  count() %>%
  ungroup() %>%
  mutate(proportion = (n / num_sites_detected$`spfko-gf`)) %>%

### plot
ggplot(aes(x = meth_change_5, y = proportion, fill = location_anno)) +
  geom_bar(stat = 'identity') +
  scale_fill_viridis_d(direction = -1, option = 'cividis') +
  facet_wrap(~ baseline_anno) +
  labs(x = 'Direction of Methylation Change',
       y = 'Proportion of Total Detectable Sites',
       fill = '',
       title = 'F') +
  coord_cartesian(ylim = c(0, 0.06)) +
  theme_classic(base_size = 20) 

### save
ggsave('spfko-gf_stacked_bar.png', width = 8, height = 8)
ggsave('spfko-gf_stacked_bar.pdf', width = 8, height = 8)
ggsave('spfko-gf_stacked_bar.tiff', width = 8, height = 8)
```

### Direction of Change by Experimental Condition Stacked Bar Plot

#### Figure G: All Sites

All significant changes by direction, hypo- or hyper- methylated for the three conditions in Figure 1, specific pathogen free vs germ free (SPF - GF) mice, IL10ko germ free mice vs germ free mice (GFKO - GF), and 

```{r}
### filter / wrange
lauren_data_wrangled %>%
  filter(sample_id %in% c('spf-gf', 'gfko-gf', 'spfko-gf'),
         meth_change_5 != 'no_change',
         significance == 'sig') %>%
  group_by(sample_id, meth_change_5) %>%
  count() %>%
  ungroup() %>% 
  mutate(proportion = n / rep(c(num_sites_detected$`spf-gf`,
                                num_sites_detected$`gfko-gf`,
                                num_sites_detected$`spfko-gf`), each = 2),
         sample_name = factor(rep(c('IL10KO', 'Microbiome', 'Both'), each = 2),
                              levels = c('Microbiome', 'IL10KO', 'Both')),
         direction = ifelse(meth_change_5 == 'hypo', 
                            'decrease', 'increase')) %>%

### plot
ggplot(aes(x = sample_name, y = proportion, fill = direction)) +
  geom_bar(stat = 'identity', position = 'dodge') +
  scale_fill_manual(values = wes_palette("Zissou1")[c(1,4)]) +
  labs(x = '',
       y = 'Proportion of\nTotal Detectable Sites',
       fill = '',
       title = 'G') +
  theme_classic(base_size = 20)

### save
ggsave('site_direction_dodged_bar.png', width = 8, height = 8)
ggsave('site_direction_dodged_bar.pdf', width = 8, height = 8)
ggsave('site_direction_dodged_bar.tiff', width = 8, height = 8)
```

#### Figure H: Sites in CpG Islands Only

```{r}
### filter / wrange
lauren_data_wrangled %>%
  filter(sample_id %in% c('spf-gf', 'gfko-gf', 'spfko-gf'),
         meth_change_5 != 'no_change',
         significance == 'sig',
         location_anno == 'CGI') %>%
  group_by(sample_id, meth_change_5) %>%
  count() %>%
  ungroup() %>% 
  mutate(proportion = n / rep(c(num_sites_detected$`spf-gf`,
                                num_sites_detected$`gfko-gf`,
                                num_sites_detected$`spfko-gf`), each = 2),
         sample_name = factor(rep(c('IL10KO', 'Microbiome', 'Both'), each = 2),
                              levels = c('Microbiome', 'IL10KO', 'Both')),
         direction = ifelse(meth_change_5 == 'hypo', 
                            'decrease', 'increase')) %>%

### plot
ggplot(aes(x = sample_name, y = proportion, fill = direction)) +
  geom_bar(stat = 'identity', position = 'dodge') +
  scale_fill_manual(values = wes_palette("Zissou1")[2-3]) +
  labs(x = '',
       y = 'Proportion of\nTotal Detectable Sites',
       fill = '',
       title = 'H') +
  theme_classic(base_size = 20)

### save
ggsave('site_direction_cgi_dodged_bar.png', width = 8, height = 8)
ggsave('site_direction_cgi_dodged_bar.pdf', width = 8, height = 8)
ggsave('site_direction_cgi_dodged_bar.tiff', width = 8, height = 8)
```

### sessionInfo

```{r}
sessionInfo()
```

<br><br>

