---
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(conflicted)
library(viridis)
library(magrittr)
library(UpSetR)
library(broom)

filter <- dplyr::filter


knitr::opts_chunk$set(echo = TRUE)
```

## More aging for 2019-05-20 data meeting




### Wrangle Data

This section copied from `analysis/figures/paperfigures.Rmd`. Should be able to slot the plot code into `paper_figures.Rmd` once a decision has been reached.

```{r}
# get file names
lauren_files <- list.files(path = '../../../data/lauren_volc_tbls', 
                           pattern = '*.csv', 
                           full.names = T)

# read files in recursively
tibble(filename = lauren_files) %>%
# read in files recursively, adding on filename
  mutate(file_contents = map(filename, ~ read_csv(.))) %>%
# expand files into skinny table
  unnest() %>%
# select only needed columns (many extra annotation columns in these tables)
  select(filename, ID, Chromosome, Mm9_SmaI_annotation_Position, 
         Ave_Y, O.Y, AvgWT, KO.WT, AvgGF, SPF.GF, AvgN, AOM.N, ttest2, pttest2, 
         inCpG, cgi.shore) -> lauren_data
```

```{r}
lauren_data %>%
# separate the filename into component folders and parts with the ultimate goal
# of using the filename to create a sample ID
  separate(filename, into = c('folder1', 'folder2', 'folder3', 'folder4', 'folder5',
                              'filename'), 
           sep = '/') %>%
# drop the folder names because I don't need them
  select(-(folder1:folder5)) %>% 
# separate the file names into their main bodies and file extensions
  separate(filename, into = c('file_prefix', 'file_extension'), sep = -4) %>% 
# drop the file extensions because I don't need them; I know they're all CSVs
  select(-file_extension) %>%
# select everything from the filename up until the first _ and use that as the
# sample ID
  mutate(sample_id = str_extract(file_prefix, "[^_]+")) %>%
# replace NAs, which are only in the methylation difference columns; this way I
# can add them together to make a unified methylation difference column rather 
# than using a complicate if or case_when statement
  replace(is.na(.), 0) %>%
# add the methylation difference columns together to make a unified methylation
# difference column
  mutate(meth_diff = O.Y + KO.WT + SPF.GF + AOM.N,
# add a categorical column annotating the direction of methylation change for
# easier counting later
         baseline_meth = Ave_Y + AvgWT + AvgGF + AvgN,
         baseline_anno = factor(case_when(baseline_meth <= 20 ~ 'Me < 20',
                                   baseline_meth > 20 & 
                                     baseline_meth < 80 ~ '20 < Me < 80',
                                   baseline_meth >= 80 ~ 'Me > 80'),
                                levels = c('Me < 20', '20 < Me < 80', 
                                           'Me > 80')),
         location_anno = case_when(inCpG == 'Y' ~ 'CGI',
                                   cgi.shore == 1 ~ 'CGI shore',
                                   TRUE ~ 'other'),
         meth_change_0 = factor(ifelse(meth_diff < 0, 'hypo', 'hyper'), 
                              levels = c('hypo', 'hyper')),
        meth_change_5 = factor(case_when(meth_diff <= -5 ~ 'hypo', 
                                         meth_diff >= 5 ~ 'hyper',
                                         TRUE ~ 'no_change'), 
                              levels = c('hypo', 'hyper', 'no_change')),
# add a categorical column annotating whether the pvalue is significant or not
# for easier counting later
         significance = ifelse(ttest2 < 0.05, 'sig', 'notsig'),
# add a categorical column annotating color for easy coloring on volcano plots
         volc_color = ifelse(abs(meth_diff) >= 5 & ttest2 < 0.05, 
                             'sig', 'notsig')) %>%
# keep only the columns I need and rename some for better intuition/easier use 
# down the line
  select(sample_id, sma_id = ID, 
         chrom = Chromosome, pos = Mm9_SmaI_annotation_Position, 
         meth_diff, pvalue = ttest2, 
         logpvalue = pttest2, meth_change_0, 
         meth_change_5, significance, 
         volc_color, baseline_meth, 
         baseline_anno, location_anno) -> lauren_data_wrangled
```

```{r}
read_tsv('../../lm/lm.tsv') -> lm
```

```{r}
# find sets tested in both analyses
lauren_data_wrangled %>%
  filter(sample_id == 'aging') %>%
  select(sma_id, term = sample_id, meth_diff, sig = pvalue) %>%
  rbind(select(lm, sma_id = mm9_SmaI_ID, term, meth_diff = estimate, 
         sig = qvalue)) %>% 
  group_by(sma_id) %>% 
  count() %>% 
  ungroup() %>% 
  filter(n == 4) -> site_whitelist
```

```{r}
# combine data together and filter for common sites test
lauren_data_wrangled %>%
  filter(sample_id == 'aging') %>%
  mutate(qvalue = NA, term = sample_id) %>%
  select(sma_id, term = sample_id, meth_diff, sig = pvalue) %>%
  rbind(select(lm, sma_id = mm9_SmaI_ID, term, meth_diff = estimate, 
         sig = qvalue)) %>%
  filter(sma_id %in% site_whitelist$sma_id) -> comb_data
```

```{r}
comb_data %>%
  filter(term != 'drug1') %>%
  dplyr::select(sma_id, term, meth_diff) %>%
  spread(term, meth_diff) %>%
  dplyr::rename(aging_meth_diff = aging, 
         inflammation_meth_diff = inflammation1,
         microbiome_meth_diff = microbiome1) -> wide_meth_diff

comb_data %>%
  filter(term != 'drug1') %>%
  dplyr::select(sma_id, term, sig) %>%
  spread(term, sig) %>%
  dplyr::rename(aging_pvalue = aging, 
         inflammation_qvalue = inflammation1,
         microbiome_qvalue = microbiome1) -> wide_sig

left_join(wide_meth_diff, wide_sig) -> wide_data
```

### Age Trend with Linear Model

#### Scatterplots

**Significant Aging Sites**

```{r}
wide_data %>%
  filter(abs(aging_meth_diff) >= 5,
         aging_pvalue < 0.05) %>%
  gather(lm_term, lm_meth_diff,
         inflammation_meth_diff:microbiome_meth_diff) %>%
  mutate(lm_term = ifelse(lm_term == 'microbiome_meth_diff',
                           'microbiome', 'inflammation')) -> age_sig
```

```{r}
### inflammation
# correlation/r value for annotating the plot
age_sig %>%
  filter(lm_term == 'inflammation') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'inflammation',
         round_value = round(estimate, 2),
         label = 'r = ',
         lm_meth_diff = 30,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  dplyr::select(lm_term, anno, lm_meth_diff, 
                aging_meth_diff) -> age_sig_inflam_cor

# pvalue for annotating the plot
age_sig %>%
  filter(lm_term == 'inflammation') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'inflammation',
         round_value = formatC(p.value, format = 'e', digits = 2),
         label = 'p = ',
         lm_meth_diff = 27,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  dplyr::select(lm_term, anno, lm_meth_diff, 
                aging_meth_diff) -> age_sig_inflam_pvalue

# slope for annotating the plot
age_sig %>%
  filter(lm_term == 'inflammation') %>%
  lm(lm_meth_diff ~ aging_meth_diff, data = .) %>%
  tidy() %>%
  filter(term != '(Intercept)') %>%
  mutate(lm_term = 'inflammation',
         round_value = round(estimate, 2),
         label = 'slope = ',
         lm_meth_diff = 24,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  dplyr::select(lm_term, anno, lm_meth_diff, 
                aging_meth_diff) -> age_sig_inflam_slope
```

```{r}
### microbiome
# correlation/r value for annotating the plot
age_sig %>%
  filter(lm_term == 'microbiome') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'microbiome',
         round_value = round(estimate, 2),
         label = 'r = ',
         lm_meth_diff = 30,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  dplyr::select(lm_term, anno, lm_meth_diff, 
                aging_meth_diff) -> age_sig_micro_cor

# pvalue for annotating the plot
age_sig %>%
  filter(lm_term == 'microbiome') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'microbiome',
         round_value = formatC(p.value, format = 'e', digits = 2),
         label = 'p = ',
         lm_meth_diff = 27,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  dplyr::select(lm_term, anno, lm_meth_diff, 
                aging_meth_diff) -> age_sig_micro_pvalue

# slope for annotating the plot
age_sig %>%
  filter(lm_term == 'microbiome') %>%
  lm(lm_meth_diff ~ aging_meth_diff, data = .) %>%
  tidy() %>%
  filter(term != '(Intercept)') %>%
  mutate(lm_term = 'microbiome',
         round_value = round(estimate, 2),
         label = 'slope = ',
         lm_meth_diff = 24,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  dplyr::select(lm_term, anno, lm_meth_diff, 
                aging_meth_diff) -> age_sig_micro_slope
```

```{r}
rbind(age_sig_inflam_cor, age_sig_inflam_pvalue, 
      age_sig_inflam_slope,
      age_sig_micro_cor, age_sig_micro_pvalue, 
      age_sig_micro_slope) -> age_sig_anno
```

```{r}
ggplot(age_sig, aes(x = aging_meth_diff, y = lm_meth_diff)) +
  geom_point(size = 1) +
  geom_smooth(method = 'lm', se = F, 
              color = rep(c('firebrick3', 'deepskyblue3'), each = 80)) +
  geom_text(data = age_sig_anno, aes(label = anno)) + 
  facet_wrap(~ lm_term) +
  labs(x = 'Change in % Methylation - Age (Old - Young)', 
     y = 'Sma Site Linear Model Slope') +
  theme_classic() +
  theme(legend.position = 'none', 
        strip.text.x = element_text(size = 12),
        axis.title = element_text(size = 16))
ggsave('age_scatterplot.png', width = 8, height = 4.5)
ggsave('age_scatterplot.pdf', width = 8, height = 4.5)
ggsave('age_scatterplot.pdf', width = 8, height = 4.5)
```

---

**Significant Microbiome/IL10 Sites (BOTH)**

```{r}
wide_data %>%
  filter(abs(inflammation_meth_diff) >= 5,
         inflammation_qvalue < 0.05,
         abs(microbiome_meth_diff) >= 5,
         microbiome_qvalue < 0.05) %>%
  gather(lm_term, lm_meth_diff,
         inflammation_meth_diff:microbiome_meth_diff) %>%
  mutate(lm_term = ifelse(lm_term == 'microbiome_meth_diff',
                           'microbiome', 
                          'inflammation')) -> micro_inflam_both_sig
```

```{r}
### inflammation
# correlation/r value for annotating the plot
micro_inflam_both_sig %>%
  filter(lm_term == 'inflammation') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'inflammation',
         round_value = round(estimate, 2),
         label = 'r = ',
         lm_meth_diff = 26,
         aging_meth_diff = 50) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_inflam_cor

# pvalue for annotating the plot
micro_inflam_both_sig %>%
  filter(lm_term == 'inflammation') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'inflammation',
         round_value = formatC(p.value, format = 'e', digits = 2),
         label = 'p = ',
         lm_meth_diff = 26,
         aging_meth_diff = 45) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_inflam_pvalue

# slope for annotateing the plot
micro_inflam_both_sig %>%
  filter(lm_term == 'inflammation') %>%
  lm(aging_meth_diff ~ lm_meth_diff, data = .) %>%
  tidy() %>%
  filter(term != '(Intercept)') %>%
  mutate(lm_term = 'inflammation',
         round_value = round(estimate, 2),
         label = 'slope = ',
         lm_meth_diff = 26,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_inflam_slope
```

```{r}
### microbiome
# correlation/r value for annotating the plot
micro_inflam_both_sig %>%
  filter(lm_term == 'microbiome') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'microbiome',
         round_value = round(estimate, 2),
         label = 'r = ',
         lm_meth_diff = 26,
         aging_meth_diff = 50) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_micro_cor

# pvalue for annotating the plot
micro_inflam_both_sig %>%
  filter(lm_term == 'microbiome') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'microbiome',
         round_value = formatC(p.value, format = 'e', digits = 2),
         label = 'p = ',
         lm_meth_diff = 26,
         aging_meth_diff = 45) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_micro_pvalue

# slope for annotating the plot
micro_inflam_both_sig %>%
  filter(lm_term == 'microbiome') %>%
  lm(aging_meth_diff ~ lm_meth_diff, data = .) %>%
  tidy() %>%
  filter(term != '(Intercept)') %>%
  mutate(lm_term = 'microbiome',
         round_value = round(estimate, 2),
         label = 'slope = ',
         lm_meth_diff = 26,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_micro_slope
```

```{r}
rbind(mi_sig_inflam_cor, mi_sig_inflam_pvalue, 
      mi_sig_inflam_slope,
      mi_sig_micro_cor, mi_sig_micro_pvalue, 
      mi_sig_micro_slope) -> micro_inflam_both_sig_anno
```

```{r}
ggplot(micro_inflam_both_sig, aes(x = lm_meth_diff, y = aging_meth_diff)) +
  geom_point() +
  geom_smooth(method = 'lm', se = F, 
              color = rep(c('firebrick3', 'deepskyblue3'), each = 80)) +
  geom_text(data = micro_inflam_both_sig_anno, aes(label = anno)) + 
  facet_wrap(~ lm_term) +
  labs(x = 'Sma Site Linear Model Slope',
       y = 'Change in % Methylation - Age (Old - Young)') +
  theme_classic() +
  theme(legend.position = 'none', 
        strip.text.x = element_text(size = 12),
        axis.title = element_text(size = 16))
ggsave('micro_inflam_sig_BOTH_scatterplot.png', width = 8, height = 4.5)
ggsave('micro_inflam_sig_BOTH_scatterplot.pdf', width = 8, height = 4.5)
ggsave('micro_inflam_sig_BOTH_scatterplot.tiff', width = 8, height = 4.5)
```

- 

**Significant Microbiome/IL10 Sites (EITHER)**

```{r}
wide_data %>%
  filter(abs(inflammation_meth_diff) >= 5 & inflammation_qvalue < 0.05 | 
         abs(microbiome_meth_diff) >= 5 & microbiome_qvalue < 0.05) %>%
  gather(lm_term, lm_meth_diff,
         inflammation_meth_diff:microbiome_meth_diff) %>%
  mutate(lm_term = ifelse(lm_term == 'microbiome_meth_diff',
                           'microbiome', 
                          'inflammation')) -> micro_inflam_either_sig
```

```{r}
### inflammation
# correlation/r value for annotating the plot
micro_inflam_either_sig %>%
  filter(lm_term == 'inflammation') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'inflammation',
         round_value = round(estimate, 2),
         label = 'r = ',
         lm_meth_diff = 26,
         aging_meth_diff = 50) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_either_inflam_cor

# pvalue for annotating the plot
micro_inflam_either_sig %>%
  filter(lm_term == 'inflammation') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'inflammation',
         round_value = formatC(p.value, format = 'e', digits = 2),
         label = 'p = ',
         lm_meth_diff = 26,
         aging_meth_diff = 45) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_either_inflam_pvalue

# slope for annotateing the plot
micro_inflam_either_sig %>%
  filter(lm_term == 'inflammation') %>%
  lm(aging_meth_diff ~ lm_meth_diff, data = .) %>%
  tidy() %>%
  filter(term != '(Intercept)') %>%
  mutate(lm_term = 'inflammation',
         round_value = round(estimate, 2),
         label = 'slope = ',
         lm_meth_diff = 26,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_either_inflam_slope
```

```{r}
### microbiome
# correlation/r value for annotating the plot
micro_inflam_either_sig %>%
  filter(lm_term == 'microbiome') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'microbiome',
         round_value = round(estimate, 2),
         label = 'r = ',
         lm_meth_diff = 26,
         aging_meth_diff = 50) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_either_micro_cor

# pvalue for annotating the plot
micro_inflam_either_sig %>%
  filter(lm_term == 'microbiome') %$%
  cor.test(lm_meth_diff, aging_meth_diff) %>%
  tidy() %>%
  mutate(lm_term = 'microbiome',
         round_value = formatC(p.value, format = 'e', digits = 2),
         label = 'p = ',
         lm_meth_diff = 26,
         aging_meth_diff = 45) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_either_micro_pvalue

# slope for annotating the plot
micro_inflam_either_sig %>%
  filter(lm_term == 'microbiome') %>%
  lm(aging_meth_diff ~ lm_meth_diff, data = .) %>%
  tidy() %>%
  filter(term != '(Intercept)') %>%
  mutate(lm_term = 'microbiome',
         round_value = round(estimate, 2),
         label = 'slope = ',
         lm_meth_diff = 26,
         aging_meth_diff = 40) %>%
  unite(anno, label, round_value, sep = '') %>%
  select(lm_term, anno, lm_meth_diff, aging_meth_diff) -> mi_sig_either_micro_slope
```

```{r}
rbind(mi_sig_either_inflam_cor, mi_sig_either_inflam_pvalue, 
      mi_sig_either_inflam_slope,
      mi_sig_either_micro_cor, mi_sig_either_micro_pvalue, 
      mi_sig_either_micro_slope) -> micro_inflam_either_sig_anno
```

```{r}
ggplot(micro_inflam_either_sig, aes(x = lm_meth_diff, y = aging_meth_diff)) +
  geom_point() +
  geom_smooth(method = 'lm', se = F, 
              color = rep(c('firebrick3', 'deepskyblue3'), each = 80)) +
  geom_text(data = micro_inflam_both_sig_anno, aes(label = anno)) + 
  facet_wrap(~ lm_term) +
  labs(x = 'Sma Site Linear Model Slope',
       y = 'Change in % Methylation - Age (Old - Young)') +
  theme_classic() +
  theme(legend.position = 'none', 
        strip.text.x = element_text(size = 12),
        axis.title = element_text(size = 16))
ggsave('micro_inflam_sig_EITHER_scatterplot.png', width = 8, height = 4.5)
ggsave('micro_inflam_sig_EITHER_scatterplot.pdf', width = 8, height = 4.5)
ggsave('micro_inflam_sig_EITHER_scatterplot.tiff', width = 8, height = 4.5)
```

#### Confusion Matrices

**Significant Aging Sites**

```{r}
age_sig %>%
  mutate(aging_direction = factor(ifelse(aging_meth_diff < 0, 'hypo', 'hyper'),
                                  levels = c('hypo', 'hyper')),
         lm_direction = factor(ifelse(lm_meth_diff< 0, 'hypo', 'hyper'),
                               levels = c('hypo', 'hyper'))) %>%
  group_by(lm_term, aging_direction, lm_direction) %>%
  count() %>%
  ungroup() -> confusion_counts_AGEsig
```

```{r}
### inflammation chi-squared
confusion_counts_AGEsig %>% 
  filter(lm_term == 'inflammation') %$%
  chisq.test(n) 
```

```{r}
### inflammation confusion matrix
confusion_counts_AGEsig %>% 
  filter(lm_term == 'inflammation') %>%


ggplot(aes(x = aging_direction, y = lm_direction, fill = n)) +
  geom_tile() +
  scale_fill_gradient('count', low = "gray95", high = "firebrick3") +
  geom_text(aes(x = aging_direction, y = lm_direction, label = n), 
            color = "black", size = 12) +
  labs(x = 'Age', y = 'IL10') +
  theme_classic() +
  theme(axis.title = element_text(size = 18),
        axis.text = element_text(size = 12),
        axis.ticks = element_blank(),
        axis.line = element_blank())
ggsave('confusion_matrix_age_sig_inflammation.png', width = 7, height = 4.5)
ggsave('confusion_matrix_age_sig_inflammation.pdf', width = 7, height = 4.5)
ggsave('confusion_matrix_age_sig_inflammation.tiff', width = 7, height = 4.5)
```

```{r}
### microbiome chi-squared
confusion_counts_AGEsig %>% 
  filter(lm_term == 'microbiome') %$%
  chisq.test(n) %>%
  tidy()
```

```{r}
### microbiome
confusion_counts_AGEsig %>% 
  filter(lm_term == 'microbiome') %>%


ggplot(aes(x = aging_direction, y = lm_direction, fill = n)) +
  geom_tile() +
  scale_fill_gradient('count', low = "gray95", high = "deepskyblue3") +
  geom_text(aes(x = aging_direction, y = lm_direction, label = n), 
            color = "black", size = 12) +
  labs(x = 'Age', y = 'Microbiome') +
  theme_classic() +
  theme(axis.title = element_text(size = 18),
        axis.text = element_text(size = 12),
        axis.ticks = element_blank(),
        axis.line = element_blank())
ggsave('confusion_matrix_age_sig_microbiome.png', width = 7, height = 4.5)
ggsave('confusion_matrix_age_sig_microbiome.pdf', width = 7, height = 4.5)
ggsave('confusion_matrix_age_sig_microbiome.tiff', width = 7, height = 4.5)
```

**Significant Microbiome/IL10 Sites (BOTH)**

```{r}
micro_inflam_both_sig %>%
  mutate(aging_direction = factor(ifelse(aging_meth_diff < 0, 'hypo', 'hyper'),
                                  levels = c('hypo', 'hyper')),
         lm_direction = factor(ifelse(lm_meth_diff< 0, 'hypo', 'hyper'),
                               levels = c('hypo', 'hyper'))) %>%
  group_by(lm_term, aging_direction, lm_direction) %>%
  count() %>%
  ungroup() -> confusion_counts_micro_inflam_BOTH_sig
```

```{r}
### inflammation chi-squared
confusion_counts_micro_inflam_BOTH_sig %>% 
  filter(lm_term == 'inflammation') %$%
  chisq.test(n) 
```

```{r}
### inflammation confusion matrix
confusion_counts_micro_inflam_BOTH_sig %>% 
  filter(lm_term == 'inflammation') %>%


ggplot(aes(x = lm_direction, y = aging_direction, fill = n)) +
  geom_tile() +
  scale_fill_gradient('count', low = "gray95", high = "firebrick3") +
  geom_text(aes(x = lm_direction, y = aging_direction, label = n), 
            color = "black", size = 12) +
  labs(x = 'IL10', y = 'Age') +
  theme_classic() +
  theme(axis.title = element_text(size = 18),
        axis.text = element_text(size = 12),
        axis.ticks = element_blank(),
        axis.line = element_blank())
ggsave('confusion_matrix_micro_inflam_BOTH_sig_inflammation.png', 
       width = 7, height = 4.5)
ggsave('confusion_matrix_micro_inflam_BOTH_sig_inflammation.pdf', 
       width = 7, height = 4.5)
ggsave('confusion_matrix_micro_inflam_BOTH_sig_inflammation.tiff', 
       width = 7, height = 4.5)
```

```{r}
### microbiome chi-squared
confusion_counts_micro_inflam_BOTH_sig %>% 
  filter(lm_term == 'microbiome') %$%
  chisq.test(n)
```

```{r}
### microbiome
confusion_counts_micro_inflam_BOTH_sig %>% 
  filter(lm_term == 'microbiome') %>%


ggplot(aes(x = lm_direction, y = aging_direction, fill = n)) +
  geom_tile() +
  scale_fill_gradient('count', low = "gray95", high = "deepskyblue3") +
  geom_text(aes(x = lm_direction, y = aging_direction, label = n), 
            color = "black", size = 12) +
  labs(x = 'Microbiome', y = 'Age') +
  theme_classic() +
  theme(axis.title = element_text(size = 18),
        axis.text = element_text(size = 12),
        axis.ticks = element_blank(),
        axis.line = element_blank())
ggsave('confusion_matrix_micro_inflam_BOTH_sig_microbiome.png', 
       width = 7, height = 4.5)
ggsave('confusion_matrix_micro_inflam_BOTH_sig_microbiome.pdf', 
       width = 7, height = 4.5)
ggsave('confusion_matrix_micro_inflam_BOTH_sig_microbiome.tiff', 
       width = 7, height = 4.5)
```

**Significant Microbiome/IL10 Sites (EIHER)**

```{r}
micro_inflam_either_sig %>%
  mutate(aging_direction = factor(ifelse(aging_meth_diff < 0, 'hypo', 'hyper'),
                                  levels = c('hypo', 'hyper')),
         lm_direction = factor(ifelse(lm_meth_diff< 0, 'hypo', 'hyper'),
                               levels = c('hypo', 'hyper'))) %>%
  group_by(lm_term, aging_direction, lm_direction) %>%
  count() %>%
  ungroup() -> confusion_counts_micro_inflam_EITHER_sig
```

```{r}
### inflammation chi-squared
confusion_counts_micro_inflam_EITHER_sig %>% 
  filter(lm_term == 'inflammation') %$%
  chisq.test(n) 
```

```{r}
### inflammation confusion matrix
confusion_counts_micro_inflam_EITHER_sig %>% 
  filter(lm_term == 'inflammation') %>%


ggplot(aes(x = lm_direction, y = aging_direction, fill = n)) +
  geom_tile() +
  scale_fill_gradient('count', low = "gray95", high = "firebrick3") +
  geom_text(aes(x = lm_direction, y = aging_direction, label = n), 
            color = "black", size = 12) +
  labs(x = 'IL10', y = 'Age') +
  theme_classic() +
  theme(axis.title = element_text(size = 18),
        axis.text = element_text(size = 12),
        axis.ticks = element_blank(),
        axis.line = element_blank())
ggsave('confusion_matrix_micro_inflam_EITHER_sig_inflammation.png', 
       width = 7, height = 4.5)
ggsave('confusion_matrix_micro_inflam_EITHER_sig_inflammation.pdf', 
       width = 7, height = 4.5)
ggsave('confusion_matrix_micro_inflam_EITHER_sig_inflammation.tiff', 
       width = 7, height = 4.5)
```

```{r}
### microbiome chi-squared
confusion_counts_micro_inflam_EITHER_sig %>% 
  filter(lm_term == 'microbiome') %$%
  chisq.test(n)
```

```{r}
### inflammation
confusion_counts_micro_inflam_EITHER_sig %>% 
  filter(lm_term == 'microbiome') %>%


ggplot(aes(x = lm_direction, y = aging_direction, fill = n)) +
  geom_tile() +
  scale_fill_gradient('count', low = "gray95", high = "deepskyblue3") +
  geom_text(aes(x = lm_direction, y = aging_direction, label = n), 
            color = "black", size = 12) +
  labs(x = 'Microbiome', y = 'Age') +
  theme_classic() +
  theme(axis.title = element_text(size = 18),
        axis.text = element_text(size = 12),
        axis.ticks = element_blank(),
        axis.line = element_blank())
ggsave('confusion_matrix_micro_inflam_EITHER_sig_microbiome.png', 
       width = 7, height = 4.5)
ggsave('confusion_matrix_micro_inflam_EITHER_sig_microbiome.pdf', 
       width = 7, height = 4.5)
ggsave('confusion_matrix_micro_inflam_EITHER_sig_microbiome.tiff', 
       width = 7, height = 4.5)
```

#