---
output: html_document
---

```{r setup}
library(tidyverse)
library(conflicted)
library(viridis)

filter <- dplyr::filter

knitr::opts_chunk$set(echo = TRUE)
```

## Microbiome Methylation Paper Figures

Code for figures for the paper The Microbiome Modulates DNA Methylation in Mouse Colonic Mucosa

### Figure 1: Affect of Microbiome and Inflammation (IL10KO) on Methylation

Volcano plots, stacked bar plots of significant CpG locations, stacked bar plot of whether sites were were significant for the microbiome, IL10KO, or both

---

#### Wrangle Data

Read in the data

```{r}
# get file names
lauren_files <- list.files(path = '../data/lauren_volc_tbls', 
                           pattern = '*.csv', 
                           full.names = T)

# read files in recursively
tibble(filename = lauren_files) %>%
# read in files recursively, adding on filename
  mutate(file_contents = map(filename, ~ read_csv(.))) %>%
# expand files into skinny table
  unnest() %>%
# select only needed columns (many extra annotation columns in these tables)
  select(filename, ID, Chromosome, Mm9_SmaI_annotation_Position, 
         Ave_Y, O.Y, AvgWT, KO.WT, AvgGF, SPF.GF, AvgN, AOM.N, ttest2, pttest2, 
         inCpG, cgi.shore) -> lauren_data
```

```{r}
lauren_data %>%
# separate the filename into component folders and parts with the ultimate goal
# of using the filename to create a sample ID
  separate(filename, into = c('folder1', 'folder2', 'folder3', 'filename'), 
           sep = '/') %>%
# drop the folder names because I don't need them
  select(-(folder1:folder3)) %>% 
# separate the file names into their main bodies and file extensions
  separate(filename, into = c('file_prefix', 'file_extension'), sep = -4) %>% 
# drop the file extensions because I don't need them; I know they're all CSVs
  select(-file_extension) %>%
# select everything from the filename up until the first _ and use that as the
# sample ID
  mutate(sample_id = str_extract(file_prefix, "[^_]+")) %>%
# replace NAs, which are only in the methylation difference columns; this way I
# can add them together to make a unified methylation difference column rather 
# than using a complicate if or case_when statement
  replace(is.na(.), 0) %>%
# add the methylation difference columns together to make a unified methylation
# difference column
  mutate(meth_diff = O.Y + KO.WT + SPF.GF + AOM.N,
# add a categorical column annotating the direction of methylation change for
# easier counting later
         baseline_meth = Ave_Y + AvgWT + AvgGF + AvgN,
         baseline_anno = factor(case_when(baseline_meth <= 20 ~ 'Me < 20',
                                   baseline_meth > 20 & 
                                     baseline_meth < 80 ~ '20 < Me < 80',
                                   baseline_meth >= 80 ~ 'Me > 80'),
                                levels = c('Me < 20', '20 < Me < 80', 
                                           'Me > 80')),
         location_anno = case_when(inCpG == 'Y' ~ 'CGI',
                                   cgi.shore == 1 ~ 'CGI shore',
                                   TRUE ~ 'other'),
         meth_change_0 = factor(ifelse(meth_diff < 0, 'hypo', 'hyper'), 
                              levels = c('hypo', 'hyper')),
        meth_change_5 = factor(case_when(meth_diff <= -5 ~ 'hypo', 
                                         meth_diff >= 5 ~ 'hyper',
                                         TRUE ~ 'no_change'), 
                              levels = c('hypo', 'hyper', 'no_change')),
# add a categorical column annotating whether the pvalue is significant or not
# for easier counting later
         significance = ifelse(ttest2 < 0.05, 'sig', 'notsig'),
# add a categorical column annotating color for easy coloring on volcano plots
         volc_color = ifelse(abs(meth_diff) >= 5 & ttest2 < 0.05, 
                             'sig', 'notsig')) %>%
# keep only the columns I need and rename some for better intuition/easier use 
# down the line
  select(sample_id, sma_id = ID, 
         chrom = Chromosome, pos = Mm9_SmaI_annotation_Position, 
         meth_diff, pvalue = ttest2, 
         logpvalue = pttest2, meth_change_0, 
         meth_change_5, significance, 
         volc_color, baseline_meth, 
         baseline_anno, location_anno) -> lauren_data_wrangled
```

#### Volcano Plots

```{r}
lauren_data_wrangled %>% distinct(sample_id)
```

**SPF - GF** Methylation differences between specific pathogen free mice and germ free mice. Contribution from having a microbiome.

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'spf-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue, color = volc_color)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_vline(xintercept = c(-5, 5), 
             color = 'gray40', 
             linetype = c('dashed')) +
  geom_hline(yintercept = -log10(0.05), 
             color = 'gray40', 
             linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (SPF - GF)',
       y = '-Log10 PValue') +
  coord_cartesian(xlim = c(-60, 60)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        strip.text.x = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        legend.position = 'none')
ggsave('spf-gf_volcano.png', width = 8, height = 8)
ggsave('spf-gf_volcano.pdf', width = 8, height = 8)
ggsave('spf-gf_volcano.tiff', width = 8, height = 8)
```

**GFIL10KO - GF** Methylation differences between germ free mice with IL10KO and germ free mice. Contribution from the IL10KO.

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'gfko-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue, color = volc_color)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_vline(xintercept = c(-5, 5), 
             color = 'gray40', 
             linetype = c('dashed')) +
  geom_hline(yintercept = -log10(0.05), 
             color = 'gray40', 
             linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (GF/IL10KO - GF)',
       y = '-Log10 PValue') +
  coord_cartesian(xlim = c(-60, 60)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        strip.text.x = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        legend.position = 'none')
ggsave('gfko-gf_volcano.png', width = 8, height = 8)
ggsave('gfko-gf_volcano.pdf', width = 8, height = 8)
ggsave('gfko-gf_volcano.tiff', width = 8, height = 8)
```

**SPFKO - GF** Methylation differences between specific pathogen free mice with IL10KO and germ free mice. Synergy between the two conditions

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'spfko-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue, color = volc_color)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_vline(xintercept = c(-5, 5), 
             color = 'gray40', 
             linetype = c('dashed')) +
  geom_hline(yintercept = -log10(0.05), 
             color = 'gray40', 
             linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (SPFKO - GF)',
       y = '-Log10 PValue') +
  coord_cartesian(xlim = c(-60, 60)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        strip.text.x = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        legend.position = 'none')
ggsave('spfko-gf_volcano.png', width = 8, height = 8)
ggsave('spfko-gf_volcano.pdf', width = 8, height = 8)
ggsave('spfko-gf_volcano.tiff', width = 8, height = 8)
```

#### Stacked CpG Location Bar Plots

```{r}
for (i in num_sites_detected$sample_id) {
  lauren_data_wrangled_2 %>%
  filter(sample_id == i, significance == 'sig') %>%
  group_by(sample_id, num_sites_detected,
           meth_change_5, location_anno, baseline_anno) %>%
  count() %>%
  ungroup() %>%
  filter(meth_change_5 != 'no_change') %>%
  mutate(proportion = (n / num_sites_detected)) %>%
   ggplot(aes(x = meth_change_5, y = proportion, fill = location_anno)) +
    geom_bar(stat = 'identity') +
    scale_fill_viridis_d(direction = -1, option = 'cividis') +
    facet_wrap(~ baseline_anno) +
    labs(x = 'Direction of Methylation Change',
         y = 'Proportion of Total Detectable Sites',
         fill = '',
         title = i) +
    theme_classic() +
    theme(axis.title = element_text(size = 16),
          strip.text.x = element_text(size = 12),
          axis.text.x = element_text(size = 12),
          legend.text = element_text(size = 12)) -> plot
  print(plot)
#  ggsave(paste('bar_cgi_prop_', i, '.png', sep = ''))
}
```





