---
output: html_document
---

```{r setup}
# general packages
library(tidyverse)
library(conflicted)
library(magrittr)
library(broom)

# plotting packages
library(viridis)
library(wesanderson)
library(UpSetR)


filter <- dplyr::filter

knitr::opts_chunk$set(echo = TRUE)
```

## Microbiome Methylation Paper Figures

Code for figures for the paper The Microbiome Modulates DNA Methylation in Mouse Colonic Mucosa

### Wrangle Data

#### Condition vs Condition Tables

These tables were created by a former graduate student in the lab, Lauren LASTNAME, who performed t-tests for experimental condition vs. experimental condition and annotated the result tables with CpG location information.

-

Read in the data

```{r}
# get file names
lauren_files <- list.files(path = '../../data/lauren_volc_tbls', 
                           pattern = '*.csv', 
                           full.names = T)

# read files in recursively
tibble(filename = lauren_files) %>%
# read in files recursively, adding on filename
  mutate(file_contents = map(filename, ~ read_csv(.))) %>%
# expand files into skinny table
  unnest() %>%
# select only needed columns (many extra annotation columns in these tables)
  select(filename, ID, Chromosome, Mm9_SmaI_annotation_Position, 
         Ave_Y, O.Y, AvgWT, KO.WT, AvgGF, SPF.GF, AvgN, AOM.N, ttest2, pttest2, 
         inCpG, cgi.shore) -> lauren_data
```

```{r}
lauren_data %>%
# separate the filename into component folders and parts with the ultimate goal
# of using the filename to create a sample ID
  separate(filename, into = c('folder1', 'folder2', 'folder3', 'folder4', 
                              'filename'), 
           sep = '/') %>%
# drop the folder names because I don't need them
  select(-(folder1:folder4)) %>% 
# separate the file names into their main bodies and file extensions
  separate(filename, into = c('file_prefix', 'file_extension'), sep = -4) %>% 
# drop the file extensions because I don't need them; I know they're all CSVs
  select(-file_extension) %>%
# select everything from the filename up until the first _ and use that as the
# sample ID
  mutate(sample_id = str_extract(file_prefix, "[^_]+")) %>%
# replace NAs, which are only in the methylation difference columns; this way I
# can add them together to make a unified methylation difference column rather 
# than using a complicate if or case_when statement
  replace(is.na(.), 0) %>%
# add the methylation difference columns together to make a unified methylation
# difference column
  mutate(meth_diff = O.Y + KO.WT + SPF.GF + AOM.N,
# add a categorical column annotating the direction of methylation change for
# easier counting later
         baseline_meth = Ave_Y + AvgWT + AvgGF + AvgN,
         baseline_anno = factor(case_when(baseline_meth <= 20 ~ 'Me < 20',
                                   baseline_meth > 20 & 
                                     baseline_meth < 80 ~ '20 < Me < 80',
                                   baseline_meth >= 80 ~ 'Me > 80'),
                                levels = c('Me < 20', '20 < Me < 80', 
                                           'Me > 80')),
         location_anno = case_when(inCpG == 'Y' ~ 'CGI',
                                   cgi.shore == 1 ~ 'CGI shore',
                                   TRUE ~ 'other'),
         meth_change_0 = factor(ifelse(meth_diff < 0, 'hypo', 'hyper'), 
                              levels = c('hypo', 'hyper')),
        meth_change_5 = factor(case_when(meth_diff <= -5 ~ 'hypo', 
                                         meth_diff >= 5 ~ 'hyper',
                                         TRUE ~ 'no_change'), 
                              levels = c('hypo', 'hyper', 'no_change')),
# add a categorical column annotating whether the pvalue is significant or not
# for easier counting later
         significance = ifelse(ttest2 < 0.05, 'sig', 'notsig'),
# add a categorical column annotating color for easy coloring on volcano plots
         volc_color = ifelse(abs(meth_diff) >= 5 & ttest2 < 0.05, 
                             'sig', 'notsig')) %>%
# keep only the columns I need and rename some for better intuition/easier use 
# down the line
  select(sample_id, sma_id = ID, 
         chrom = Chromosome, pos = Mm9_SmaI_annotation_Position, 
         meth_diff, pvalue = ttest2, 
         logpvalue = pttest2, meth_change_0, 
         meth_change_5, significance, 
         volc_color, baseline_meth, 
         baseline_anno, location_anno) -> lauren_data_wrangled
```

#### Linear Model

```{r}
read_tsv('../lm/lm.tsv') -> lm
```

### Figure 1: Affect of Microbiome and Inflammation (IL10KO) on Methylation

Volcano plots, stacked bar plots of significant CpG locations, stacked bar plot of whether sites were were significant for the microbiome, IL10KO, or both

---

#### Volcano Plots

**SPF - GF** Methylation differences between specific pathogen free mice and germ free mice. Contribution from having a microbiome.

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'spf-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue, color = volc_color)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_vline(xintercept = c(-5, 5), 
             color = 'gray40', 
             linetype = c('dashed')) +
  geom_hline(yintercept = -log10(0.05), 
             color = 'gray40', 
             linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (SPF - GF)',
       y = '-Log10 PValue') +
  coord_cartesian(xlim = c(-50, 50),
                  ylim = c(0, 10)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        legend.position = 'none')
ggsave('figure1/spf-gf_volcano.png', width = 8, height = 8)
ggsave('figure1/spf-gf_volcano.pdf', width = 8, height = 8)
ggsave('figure1/spf-gf_volcano.tiff', width = 8, height = 8)
```

**GFIL10KO - GF** Methylation differences between germ free mice with IL10KO and germ free mice. Contribution from the IL10KO.

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'gfko-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue, color = volc_color)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_vline(xintercept = c(-5, 5), 
             color = 'gray40', 
             linetype = c('dashed')) +
  geom_hline(yintercept = -log10(0.05), 
             color = 'gray40', 
             linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (GF/IL10KO - GF)',
       y = '-Log10 PValue') +
  coord_cartesian(xlim = c(-50, 50),
                  ylim = c(0, 10)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        legend.position = 'none')
ggsave('figure1/gfko-gf_volcano.png', width = 8, height = 8)
ggsave('figure1/gfko-gf_volcano.pdf', width = 8, height = 8)
ggsave('figure1/gfko-gf_volcano.tiff', width = 8, height = 8)
```

**SPFKO - GF** Methylation differences between specific pathogen free mice with IL10KO and germ free mice. Synergy between the two conditions

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'spfko-gf') %>%


ggplot(aes(x = meth_diff, y = logpvalue, color = volc_color)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_vline(xintercept = c(-5, 5), 
             color = 'gray40', 
             linetype = c('dashed')) +
  geom_hline(yintercept = -log10(0.05), 
             color = 'gray40', 
             linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (SPFKO - GF)',
       y = '-Log10 PValue') +
  coord_cartesian(xlim = c(-50, 50),
                  ylim = c(0, 10)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        legend.position = 'none')
ggsave('figure1/spfko-gf_volcano.png', width = 8, height = 8)
ggsave('figure1/spfko-gf_volcano.pdf', width = 8, height = 8)
ggsave('figure1/spfko-gf_volcano.tiff', width = 8, height = 8)
```

#### Stacked CpG Location Bar Plots

Count number of sites detected in each sample.

```{r}
lauren_data_wrangled %>%
  group_by(sample_id) %>%
  count() %>%
  ungroup() %>%
  rename(num_sites_detected = n) %>%
  spread(sample_id, num_sites_detected) -> num_sites_detected
```

---

**SPF - GF** Location of significant CpG sites from the volcano plot between specific pathogen free mice and germ free mice. 

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'spf-gf',
         significance == 'sig',
         meth_change_5 != 'no_change') %>%
  group_by(significance, meth_change_5, location_anno, baseline_anno) %>%
  count() %>%
  ungroup() %>%
  mutate(proportion = (n / num_sites_detected$`spf-gf`)) %>%


ggplot(aes(x = meth_change_5, y = proportion, fill = location_anno)) +
  geom_bar(stat = 'identity') +
  scale_fill_viridis_d(direction = -1, option = 'cividis') +
  facet_wrap(~ baseline_anno) +
  labs(x = 'Direction of Methylation Change',
       y = 'Proportion of Total Detectable Sites',
       fill = '') +
  coord_cartesian(ylim = c(0, 0.06)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        strip.text.x = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        legend.text = element_text(size = 12))
ggsave('figure1/spf-gf_stacked_bar.png', width = 8, height = 8)
ggsave('figure1/spf-gf_stacked_bar.pdf', width = 8, height = 8)
ggsave('figure1/spf-gf_stacked_bar.tiff', width = 8, height = 8)
```

**GFKO - GF** Location of significant CpG sites from the volcano plot between germ free mice with IL10KO and germ free mice. 

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'gfko-gf',
         significance == 'sig',
         meth_change_5 != 'no_change') %>%
  group_by(significance, meth_change_5, location_anno, baseline_anno) %>%
  count() %>%
  ungroup() %>%
  mutate(proportion = (n / num_sites_detected$`gfko-gf`)) %>%


ggplot(aes(x = meth_change_5, y = proportion, fill = location_anno)) +
  geom_bar(stat = 'identity') +
  scale_fill_viridis_d(direction = -1, option = 'cividis') +
  facet_wrap(~ baseline_anno) +
  labs(x = 'Direction of Methylation Change',
       y = 'Proportion of Total Detectable Sites',
       fill = '') +
  coord_cartesian(ylim = c(0, 0.06)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        strip.text.x = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        legend.text = element_text(size = 12))
ggsave('figure1/gfko-gf_stacked_bar.png', width = 8, height = 8)
ggsave('figure1/gfko-gf_stacked_bar.pdf', width = 8, height = 8)
ggsave('figure1/gfko-gf_stacked_bar.tiff', width = 8, height = 8)
```

**SPFKO - GF** Location of significant CpG sites from the volcano plot between specific pathogen free mice with IL10KO and germ free mice. 

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'spfko-gf',
         significance == 'sig',
         meth_change_5 != 'no_change') %>%
  group_by(significance, meth_change_5, location_anno, baseline_anno) %>%
  count() %>%
  ungroup() %>%
  mutate(proportion = (n / num_sites_detected$`spfko-gf`)) %>%


ggplot(aes(x = meth_change_5, y = proportion, fill = location_anno)) +
  geom_bar(stat = 'identity') +
  scale_fill_viridis_d(direction = -1, option = 'cividis') +
  facet_wrap(~ baseline_anno) +
  labs(x = 'Direction of Methylation Change',
       y = 'Proportion of Total Detectable Sites',
       fill = '') +
  coord_cartesian(ylim = c(0, 0.06)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        strip.text.x = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        legend.text = element_text(size = 12))
ggsave('figure1/spfko-gf_stacked_bar.png', width = 8, height = 8)
ggsave('figure1/spfko-gf_stacked_bar.pdf', width = 8, height = 8)
ggsave('figure1/spfko-gf_stacked_bar.tiff', width = 8, height = 8)
```

#### Direction of Change by Experimental Condition Stacked Bar Plot for SPFKO - GF

```{r}
lauren_data_wrangled %>%
  filter(sample_id %in% c('spf-gf', 'gfko-gf', 'spfko-gf'),
         meth_change_5 != 'no_change',
         significance == 'sig') %>%
  group_by(sample_id, meth_change_5) %>%
  count() %>%
  ungroup() %>% 
  mutate(proportion = n / rep(c(num_sites_detected$`spf-gf`,
                                num_sites_detected$`gfko-gf`,
                                num_sites_detected$`spfko-gf`), each = 2),
         sample_name = factor(rep(c('IL10KO', 'Microbiome', 'Both'), each = 2),
                              levels = c('Microbiome', 'IL10KO', 'Both')),
         direction = ifelse(meth_change_5 == 'hypo', 
                            'decrease', 'increase')) %>%

ggplot(aes(x = sample_name, y = proportion, fill = direction)) +
  geom_bar(stat = 'identity', position = 'dodge') +
  scale_fill_manual(values = wes_palette("Zissou1")[c(1,4)]) +
  labs(x = '',
       y = 'Proportion of Total Detectable Sites',
       fill = '') +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        axis.text.x = element_text(size = 16),
        legend.text = element_text(size = 12))
ggsave('figure1/site_direction_dodged_bar.png', width = 8, height = 8)
ggsave('figure1/site_direction_dodged_bar.pdf', width = 8, height = 8)
ggsave('figure1/site_direction_dodged_bar.tiff', width = 8, height = 8)
```

### Figure 2: Effect of Azoxymethane on Methylation

#### Volcano Plot

**SPFAOM - SPF** Methylation differences between specific pathogen free mice treated with azoxymethane and specific pathogen free mice. Effect of the drug.

```{r}
lauren_data_wrangled %>% 
  filter(sample_id == 'spfaom-spf') %>%


ggplot(aes(x = meth_diff, y = logpvalue, color = volc_color)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_vline(xintercept = c(-5, 5), 
             color = 'gray40', 
             linetype = c('dashed')) +
  geom_hline(yintercept = -log10(0.05), 
             color = 'gray40', 
             linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (SPFAOM - SPF)',
       y = '-Log10 PValue') +
  coord_cartesian(xlim = c(-50, 50),
                  ylim = c(0, 10)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        legend.position = 'none')
ggsave('figure2/spfaom-spf_volcano.png', width = 8, height = 8)
ggsave('figure2/spfaom-spf_volcano.pdf', width = 8, height = 8)
ggsave('figure2/spfaom-spf_volcano.tiff', width = 8, height = 8)
```

#### Stacked CpG Location Bar Plots

**SPFAOM - SPF** Location of significant CpG sites from the volcano plot between specific pathogen free mice treated with azoxymethane and specific pathogen free mice. 

```{r}
lauren_data_wrangled %>%
  filter(sample_id == 'spfaom-spf',
         significance == 'sig',
         meth_change_5 != 'no_change') %>%
  group_by(significance, meth_change_5, location_anno, baseline_anno) %>%
  count() %>%
  ungroup() %>%
  mutate(proportion = (n / num_sites_detected$`spfko-gf`)) %>%


ggplot(aes(x = meth_change_5, y = proportion, fill = location_anno)) +
  geom_bar(stat = 'identity') +
  scale_fill_viridis_d(direction = -1, option = 'cividis') +
  facet_wrap(~ baseline_anno) +
  labs(x = 'Direction of Methylation Change',
       y = 'Proportion of Total Detectable Sites',
       fill = '') +
  coord_cartesian(ylim = c(0, 0.06)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        strip.text.x = element_text(size = 12),
        axis.text.x = element_text(size = 12),
        legend.text = element_text(size = 12))
ggsave('figure2/spfaom-spf_stacked_bar.png', width = 8, height = 8)
ggsave('figure2/spfaom-spf_stacked_bar.pdf', width = 8, height = 8)
ggsave('figure2/spfaom-spf_stacked_bar.tiff', width = 8, height = 8)
```

### Figure 3: Linear Model

#### Volcano Plots

Wrangle the data for plotting a volcano plot

```{r}
lm %>%
# log the qvalues for easier volcano plotting
  mutate(logqvalue = -log10(qvalue), 
# annotate the direction of change to count sites for volcano plot annotation
         change = ifelse(estimate < 0, 'hypo', 'hyper'),
# add a categorical column for easy volcano plot coloring
         volc_color = case_when((qvalue < 0.05 & abs(estimate) >= 5) & 
                                  term == 'inflammation1' ~ 'sig_inf',
                                (qvalue < 0.05 & abs(estimate) >= 5) & 
                                  term == 'microbiome1' ~ 'sig_mic',
                                (qvalue < 0.05 & abs(estimate) >= 5) & 
                                  term == 'drug1' ~ 'sig_drug',
                                TRUE ~ 'notsig'),
# add a categorical column for better facet names
         facets = case_when(term == 'inflammation1' ~ 'IL10KO',
                            term == 'microbiome1' ~ 'Microbiome',
                            term == 'drug1' ~ 'Azoxymethane')) -> lm_volc
```

Count numbers of hypo/hyper methylated Sma sites to annotate the volcano plot with.

```{r}
### volcano plot annotation
lm_volc %>% 
  group_by(facets, change, volc_color) %>% 
  count() %>% 
  ungroup() %>%
  filter(volc_color != 'notsig') %>%
  mutate(perc = round(((n / 12566) * 100), 2),
         estimate = rep(c(20, -20), 3), logqvalue = 12, 
         term = rep(c('Azoxymethane', 'IL10KO', 'Microbiome'), each = 2),
         sign = '%') %>%
  unite(anno, n, perc, sep = ', ') %>%
  unite(anno, anno, sign, sep = '') -> lm_volc_counts
```

Plot the volcano plot

```{r}
ggplot(lm_volc, aes(x = estimate, y = logqvalue)) + 
  geom_point(aes(color = volc_color), size = 1) + 
  scale_color_manual(values = c('gray60', 'darkorange2', 
                                'firebrick3', 'deepskyblue3')) +
  geom_hline(yintercept = -log10(0.05), 
             linetype = 'dashed', 
             color = 'gray30') +
  geom_vline(xintercept = c(-5, 5), 
             linetype = 'dashed', 
             color = 'gray30') +
  geom_text(data = lm_volc_counts, aes(label = anno)) +
  labs(x = 'Sma Site Linear Model Slope', 
       y = '-Log10 QValue') +
  facet_wrap(. ~ facets) +
  theme_classic() +
  theme(legend.position = 'none', 
        strip.text.x = element_text(size = 12),
        axis.title = element_text(size = 16),
        axis.text.x = element_text(size = 12))
ggsave('figure3/lm_slope_volc.png', width = 8, height = 6)
ggsave('figure3/lm_slope_volc.pdf', width = 8, height = 6)
ggsave('figure3/lm_slope_volc.tiff', width = 8, height = 6)
```

#### UpSet Plots

Wrangle the data for the UpSet plot. The UpSet plot needs binary columns, 0/1, indicating whether the sma site is significant under that condition

```{r}
### hypOmethylated sites
lm_volc %>%
  filter(abs(estimate) >= 5) %>%
  mutate(term_change = case_when(term == 'microbiome1' & 
                                   estimate <= -5 ~ 'Microbiome_hypo',
                                 term == 'inflammation1' & 
                                   estimate <= -5 ~ 'IL10KO_hypo',
                                 term == 'drug1' & 
                                   estimate <= -5 ~ 'Azoxymethane_hypo',
                                 TRUE ~ 'hyper')) %>%
  filter(term_change != 'hyper') %>%
  select(mm9_SmaI_ID, term_change, qvalue) %>%
  spread(term_change, qvalue) %>%
  replace(is.na(.), 1) %>%
  mutate(Azoxymethane = ifelse(Azoxymethane_hypo < 0.05, 1, 0),
         IL10KO = ifelse(IL10KO_hypo < 0.05, 1, 0),
         Microbiome = ifelse(Microbiome_hypo < 0.05, 1, 0)) %>%
  as.data.frame() -> lm_upset_hypo

### hypERmethylated sites
lm_volc %>%
  filter(abs(estimate) >= 5) %>%
  mutate(term_change = case_when(term == 'microbiome1' & 
                                   estimate >= 5 ~ 'Microbiome_hyper',
                                 term == 'inflammation1' & 
                                   estimate >= 5 ~ 'IL10KO_hyper',
                                 term == 'drug1' & 
                                   estimate >= 5 ~ 'Azoxymethane_hyper',
                                 TRUE ~ 'hypo')) %>%
  filter(term_change != 'hypo') %>%
  select(mm9_SmaI_ID, term_change, qvalue) %>%
  spread(term_change, qvalue) %>%
  replace(is.na(.), 1) %>%
  mutate(Azoxymethane = ifelse(Azoxymethane_hyper < 0.05, 1, 0),
         IL10KO = ifelse(IL10KO_hyper < 0.05, 1, 0),
         Microbiome = ifelse(Microbiome_hyper < 0.05, 1, 0),) %>%
  as.data.frame() -> lm_upset_hyper
```

Plot and save the UpSet plots

```{r}
### significant hypOmethylated sites
#ggsave('figure3/lm_upset_hypo.png', width = 8, height = 6,
#ggsave('figure3/lm_upset_hypo.pdf', width = 8, height = 6,
#ggsave('figure3/lm_upset_hypo.tiff', width = 8, height = 6,
upset(lm_upset_hypo, 
      sets = c('Microbiome', 'IL10KO', 'Azoxymethane'),
      keep.order = T,
      order.by = 'degree',
      main.bar.color = 'dodgerblue',
      mainbar.y.label = '',
      text.scale = c(1, 2, 1, 1.75, 2, 3))
)

### significant hypERmethylated sites
ggsave('figure3/lm_upset_hyper.png', width = 8, height = 6,
#ggsave('figure3/lm_upset_hyper.png', width = 8, height = 6,
#ggsave('figure3/lm_upset_hyper.tiff', width = 8, height = 6,
upset(lm_upset_hyper, 
      sets = c('Microbiome', 'IL10KO', 'Azoxymethane'),
      keep.order = T,
      order.by = 'degree',
      main.bar.color = 'firebrick3',
      mainbar.y.label = '',
      text.scale = c(1, 2, 1, 1.75, 2, 3))
)
```

### Figure 4: Relationship between Aging and the Other Variables

#### Volcano Plot

**Old - Young** Volcano plot of methylation differences due to age, old - young

```{r}
lauren_data_wrangled %>% 
  filter(sample_id == 'aging') %>%


ggplot(aes(x = meth_diff, y = logpvalue, color = volc_color)) +
  geom_point(size = 1) +
  scale_color_manual(values = c('gray60', 'darkcyan')) +
  geom_vline(xintercept = c(-5, 5), 
             color = 'gray40', 
             linetype = c('dashed')) +
  geom_hline(yintercept = -log10(0.05), 
             color = 'gray40', 
             linetype = 'dashed') +
  labs(x = 'Difference in % Methylation (Old - Young)',
       y = '-Log10 PValue') +
  coord_cartesian(xlim = c(-50, 50),
                  ylim = c(0, 10)) +
  theme_classic() +
  theme(axis.title = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        legend.position = 'none')
ggsave('figure4/aging_volcano.png', width = 8, height = 8)
ggsave('figure4/aging_volcano.pdf', width = 8, height = 8)
ggsave('figure4/aging_volcano.tiff', width = 8, height = 8)
```

#### Wrangle Data

Combine the aging data with the linear model data

```{r}
# find sets tested in both analyses
lauren_data_wrangled %>%
  filter(sample_id == 'aging') %>%
  select(sma_id, term = sample_id, meth_diff, sig = pvalue) %>%
  rbind(select(lm, sma_id = mm9_SmaI_ID, term, meth_diff = estimate, 
         sig = qvalue)) %>% 
  group_by(sma_id) %>% 
  count() %>% 
  ungroup() %>% 
  filter(n == 4) -> site_whitelist
```

```{r}
# combine data together and filter for common sites test
lauren_data_wrangled %>%
  filter(sample_id == 'aging') %>%
  select(sma_id, term = sample_id, meth_diff, sig = pvalue) %>%
  rbind(select(lm, sma_id = mm9_SmaI_ID, term, meth_diff = estimate, 
         sig = qvalue)) %>%
  filter(sma_id %in% site_whitelist$sma_id) -> comb_data
```

```{r}
# get wide methylation and significance values
comb_data %>%
  filter(term != 'drug1') %>%
  dplyr::select(sma_id, term, meth_diff) %>%
  spread(term, meth_diff) %>%
  dplyr::rename(aging_meth_diff = aging, 
         inflammation_meth_diff = inflammation1,
         microbiome_meth_diff = microbiome1) -> wide_meth_diff

comb_data %>%
  filter(term != 'drug1') %>%
  dplyr::select(sma_id, term, sig) %>%
  spread(term, sig) %>%
  dplyr::rename(aging_pvalue = aging, 
         inflammation_qvalue = inflammation1,
         microbiome_qvalue = microbiome1) -> wide_sig

left_join(wide_meth_diff, wide_sig) -> wide_data
```

#### UpSet Plots

Annotate data for upset plots

```{r}
### hypOmethylated sites
comb_data %>% 
  filter(abs(meth_diff) >= 5, sig < 0.05) %>% 
  dplyr::select(sma_id, term, meth_diff) %>%
  spread(term, meth_diff) %>%
  mutate(Azoxymethane = ifelse(drug1 <= -5, 1, 0),
         IL10KO = ifelse(inflammation1 <= -5, 1, 0),
         Microbiome = ifelse(microbiome1 <= -5, 1, 0),
         Age = ifelse(aging <= -5, 1, 0)) %>%
  replace(is.na(.), 0) %>%
  select(sma_id, Azoxymethane:Age) %>%
  as.data.frame() -> age_upset_hypo

### hypERmethylated sites
comb_data %>% 
  filter(abs(meth_diff) >= 5, sig < 0.05) %>% 
  dplyr::select(sma_id, term, meth_diff) %>%
  spread(term, meth_diff) %>%
  mutate(Azoxymethane = ifelse(drug1 >= 5, 1, 0),
         IL10KO = ifelse(inflammation1 >= 5, 1, 0),
         Microbiome = ifelse(microbiome1 >= 5, 1, 0),
         Age = ifelse(aging >= 5, 1, 0)) %>%
  replace(is.na(.), 0) %>%
  select(sma_id, Azoxymethane:Age) %>%
  as.data.frame() -> age_upset_hyper
```

```{r}
### hypOmethylated sites
ggsave('figure4/age_upset_hypo.png', width = 8, height = 6,
#ggsave('figure4/age_upset_hypo.pdf', width = 8, height = 6,
#ggsave('figure4/age_upset_hypo.tiff', width = 8, height = 6,
upset(age_upset_hypo,
      sets = c('Age', 'Microbiome', 'IL10KO', 'Azoxymethane'),
      keep.order = T,
      order.by = 'degree',
      main.bar.color = 'dodgerblue3',
      mainbar.y.label = '',
      text.scale = c(1, 2, 1, 2, 2, 2))
)

### hypERmethylated sites
ggsave('figure4/age_upset_hyper.png', width = 8, height = 6,
#ggsave('figure4/age_upset_hyper.pdf', width = 8, height = 6,
#ggsave('figure4/age_upset_hyper.tiff', width = 8, height = 6,
upset(age_upset_hyper,
      sets = c('Age', 'Microbiome', 'IL10KO', 'Azoxymethane'),
      keep.order = T,
      order.by = 'degree',
      main.bar.color = 'firebrick4',
      mainbar.y.label = '',
      text.scale = c(1, 2, 1, 2, 2, 2))
)
```

#### Scatter Plots

*2019-05-29* - See `temp_scatterplot_confusion_matrics.Rmd` for code until we decide what version of the scatter plots/confusion matrices we're going with